lib/versions.ps1

```diff
@@ -1,53 +1,231 @@
 # versions
-function latest_version($app, $bucket, $url) {
-    (manifest $app $bucket $url).version
-}
-function current_version($app, $global) {
-    @(versions $app $global)[-1]
+function Get-LatestVersion {
+    <#
+    .SYNOPSIS
+        Get latest version of app
+    .DESCRIPTION
+        Get latest version of app from manifest
+    .PARAMETER App
+        App's name
+    .PARAMETER Bucket
+        Bucket which the app belongs to
+    .PARAMETER URL
+        Remote app manifest's URI
+    #>
+    [OutputType([String])]
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $true, Position = 0, ValueFromPipeline = $true)]
+        [String]
+        $App,
+        [Parameter(Position = 1)]
+        [String]
+        $Bucket,
+        [Parameter(Position = 2)]
+        [String]
+        $URL
+    )
+    return (manifest $App $Bucket $URL).version
 }
-function versions($app, $global) {
-    $appdir = appdir $app $global
-    if(!(test-path $appdir)) { return @() }
 
-    sort_versions (Get-ChildItem $appdir -dir -attr !reparsePoint | Where-Object { $null -ne $(Get-ChildItem $_.fullname) } | ForEach-Object { $_.name })
+function Select-CurrentVersion {
+    <#
+    .SYNOPSIS
+        Select current version of app
+    .DESCRIPTION
+        Select current version of installed app, from 'current\manifest.json' or modified time of version directory
+    .PARAMETER App
+        App's name
+    .PARAMETER Global
+        Globally installed application
+    #>
+    [OutputType([String])]
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory = $true, Position = 0, ValueFromPipeline = $true)]
+        [String]
+        $App,
+        [Parameter(Position = 1)]
+        [Switch]
+        $Global
+    )
+
+    $appPath = appdir $App $Global
+    if (Test-Path "$appPath\current") {
+        $currentVersion = (installed_manifest $App 'current' $Global).version
+    } else {
+        $installedVersion = Get-InstalledVersion -App $App -Global:$Global
+        if ($installedVersion) {
+            $currentVersion = $installedVersion[-1]
+        } else {
+            $currentVersion = $null
+        }
+    }
+    return $currentVersion
 }
 
-function version($ver) {
-    $ver -split '[\.-]' | ForEach-Object {
-        $num = $_ -as [int]
-        if($num) { $num } else { $_ }
+function Get-InstalledVersion {
+    <#
+    .SYNOPSIS
+        Get installed version of app
+    .DESCRIPTION
+        Get all installed version of app, by checking version directories' 'install.json'
```

Comment:
Add note about sort order. Without it #L56 is pretty confusing without looking at code